### ISSUE TITLE
Host pattern foo[1:3] is sometimes not recognized

### ISSUE THREAD

**Issue by jmaschtaler** [ansible/ansible#23393]
- Bug Report

##### COMPONENT NAME
Pattern

##### ANSIBLE VERSION
```
ansible 2.2.2.0
  config file = /home/xxx/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
Ubuntu 14.04.05

##### SUMMARY
Host pattern foo0[1:3] does not recognize all hosts under all circumstances.

##### STEPS TO REPRODUCE
**First way to reproduce:**
1. Create hosts foo0[1:3].domain.tld
2. Create inventory with entry:
```
[group]
foo0[1:3].domain.tld
```
3. Run `ansible -k -u root -i inventory -m shell -a 'uptime' 'foo0[1:3].domain.tld'`

**Second way to reproduce:**

4. Create playbook.yaml:
```
- --
- hosts: 'foo0[1:3].domain.tld'
  gather_facts: false

  tasks:
    - name: 'uptime'
      shell: '/usr/bin/uptime'
```
5. Run `ansible-playbook -u root -i inventory playbook.yaml`

##### EXPECTED RESULTS
```
foo01.domain.tld | SUCCESS | rc=0 >>
 12:23:50 up 30 days, 22:34,  1 user,  load average: 0.16, 0.14, 0.09

foo02.domain.tld | SUCCESS | rc=0 >>
 12:23:50 up 30 days, 22:42,  1 user,  load average: 0.26, 0.28, 0.27

foo03.domain.tld | SUCCESS | rc=0 >>
 12:23:50 up 30 days, 22:15,  1 user,  load average: 0.14, 0.08, 0.0
```

##### ACTUAL RESULTS
```
foo01.domain.tld | SUCCESS | rc=0 >>
 12:22:17 up 30 days, 22:33,  1 user,  load average: 0.15, 0.15, 0.09

foo03.domain.tld | SUCCESS | rc=0 >>
 12:22:17 up 30 days, 22:40,  1 user,  load average: 0.15, 0.28, 0.26
```

Addition:
With the playbook and inventory created above, this command gives the expected result:
`ansible -u root -i inventory -m shell -a 'uptime' 'group'`

**Comment by snowsky** [comment]
cannot reproduce for 2.2.1.0 and 2.4(devel 13b2f11139)

**Comment by jmaschtaler** [comment]
Hi,

thanks for response and checking.

However we can still reproduce it, also on
- Archlinux (2.2.2.0)
- Mac (2.2.1.0)

Maybe the pattern _0[1:3]_ is not supposed to be used in the commandline and playbook?

Best,
jörg

**Comment by jmaschtaler** [comment]
It's fixed with version 2.3.0.0
